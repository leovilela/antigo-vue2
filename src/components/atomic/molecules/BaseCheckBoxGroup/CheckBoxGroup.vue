<template>
  <div :class="formClass">
    <Base-Checkbox v-on="listeners" v-bind="$attrs" :value="value" />
    <label for="" v-if="label">{{label}}</label>
    <slot />
  </div>
</template>

<script>
import FormMixing from "../../mixins/form";

export default {
  mixins: [FormMixing],
  props: {
    formClass: {
      type: Object,
      default: () => {},
    },
    label: {
      type: String,
      default: null,
    },
    value: [String, Number, Boolean, Function, Object, Array],
  },
};
</script>
