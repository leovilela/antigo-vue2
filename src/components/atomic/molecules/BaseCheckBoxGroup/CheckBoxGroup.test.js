import { shallowMount } from "@vue/test-utils";
import CheckBoxGroup from "../BaseCheckBoxGroup/CheckBoxGroup";

const factory = () => shallowMount(CheckBoxGroup, {
  stubs: ["Base-Checkbox"],
});

describe("CheckBoxGroup", () => {
  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
