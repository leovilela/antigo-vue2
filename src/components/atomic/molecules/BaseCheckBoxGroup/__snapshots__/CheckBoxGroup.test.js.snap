// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxGroup renders properly 1`] = `
<div>
  <base-checkbox-stub></base-checkbox-stub>
  <!---->
</div>
`;
